operator/v1/types_network.go

```diff
@@ -53,6 +53,7 @@ type NetworkList struct {
 
 // NetworkSpec is the top-level network configuration object.
 // +kubebuilder:validation:XValidation:rule="!has(self.defaultNetwork) || !has(self.defaultNetwork.ovnKubernetesConfig) || !has(self.defaultNetwork.ovnKubernetesConfig.gatewayConfig) || !has(self.defaultNetwork.ovnKubernetesConfig.gatewayConfig.ipForwarding) || self.defaultNetwork.ovnKubernetesConfig.gatewayConfig.ipForwarding == oldSelf.defaultNetwork.ovnKubernetesConfig.gatewayConfig.ipForwarding || self.defaultNetwork.ovnKubernetesConfig.gatewayConfig.ipForwarding == 'Restricted' || self.defaultNetwork.ovnKubernetesConfig.gatewayConfig.ipForwarding == 'Global'",message="invalid value for IPForwarding, valid values are 'Restricted' or 'Global'"
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=AdvancedRoutingBGP,rule="(has(self.advancedRouting) && in('FRR',self.advancedRouting.providers)) || !has(self.defaultNetwork.ovnKubernetesConfig) || self.defaultNetwork.ovnKubernetesConfig.routeAdvertisements != 'Enabled'",message="Route advertisements cannot be Enabled if FRR advanced routing provider is not available"
```

Comment:
Does this mean the featuregate must be enabled to deploy frr-k8s alone? Asking because we might need from MetalLB regardless of the fact that the feature gate is open.

Reply:
The feature gate is a requirement before graduating api changes to GA. Once the feature is tested before GA, we can remove the feature gate.